lib/transaction.js

```diff
@@ -225,12 +203,11 @@ class Transaction extends EventEmitter {
       .then((connection) => {
         connection.__knexTxId = this.txid;
 
-        return (this._previousSibling
-          ? this._previousSibling.catch(() => {})
-          : Promise.resolve()
-        ).then(function() {
-          return connection;
-        });
+        return this._previousSibling
+          // .catch(() => {}) // TODO: Investigate this line
+          .then(function() {
```

Comment:
probably can be replaced with async/await?

Reply:
It's worth a shot, at least.  My main concern was that `Transaction` itself is a Promise-like object.  So, we need to make sure it isn't partially-resolved by accident.

Reply:
Actually: Just realized that this PR is showing items that were merged from the other PR (#3671 ) .  I might need to rebase the branch and create a new PR instead.